Makefile

```diff
@@ -20,10 +20,16 @@ e2e-test: otelcontribcol
 	$(MAKE) -C testbed runtests
 
 .PHONY: ci
-ci: all test-with-cover
+ci: gomod-check all test-with-cover
 	$(MAKE) -C testbed install-tools
 	$(MAKE) -C testbed runtests
 
+.PHONY: gomod-check
+gomod-check:
+	@go mod download
+	$(MAKE) gotidy
+	@git diff --exit-code || (echo 'Go modules are not tidied up. Run `make gotidy` and commit the changes.' && exit 1)
```

Comment:
I am not sure about this one. `go mod tidy` followed by `make` often ends up modifying go.mod/go.sum files. `make` is usually the last thing I run before creating a commit. This requires that `go mod tidy` is the last thing I do.

Reply:
Running `make` right after `go mod tidy` should never result in any changes to go.mod/sum files. If it does then there is either a bug in the go command or we've setup modules incorrectly somehow. May be the tools thing is interfering? I'll try to reproduce the case.